### PR TITLE
Added:Login functionality

### PR DESCRIPTION
**Pull request by siddhishinde0723**
Added login functionality 

## Summary by CodeRabbit

- **New Features**
	- Enhanced the login form to support additional fields for a comprehensive authentication process.
	- Introduced a logout button with an icon and localized text for logout and settings in the topbar.
	- Implemented internationalization for dynamic localization in the footer text and login-related labels.

- **Improvements**
	- Updated the `LoginComponent` to enhance user login functionality, including improved form handling and session storage management.
	- Added new constants and configurations for better session management and authentication process.

- **Localization**
	- Expanded localization support with additional login-related labels and messages in English.

### CODE REVIEW

File: src/app/sb-admin/components/login/login.component.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  private createRequestBody(values: any, targetUrl: string): URLSearchParams {
+    const body = new URLSearchParams();
+    body.set('client_id', 'implementation');
+    body.set('client_secret', values.clientSecret);
+    body.set('grant_type', 'password');
+    body.set('username', values.userName);
+    body.set('password', values.Password);
+
+    return body;
+  }
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/app/sb-admin/components/login/login.component.ts]: fix this @siddhishinde0723

File: src/app/sb-admin/components/pages/user-dashboard/user-dashboard.component.html [hunk 2]
```diff
@@ -72,7 +72,7 @@ <h5 class="m-0">{{ 'COMMON_USERS' | i18next}}</h5><br>
                         </td>
                     </tr>
                 </ng-template>
-            </p-table>
+            </p-table>login
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/app/sb-admin/components/pages/user-dashboard/user-dashboard.component.html]: fix this @siddhishinde0723

File: src/app/sb-admin/components/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      authToken: ['', Validators.required],
+      userName: ['', Validators.required],
+      Password: ['', Validators.required],
+      targetURL: ['', Validators.required],
+      clientSecret:['',Validators.required]
+    });
+  }
+
+
+
```
Comment (manojLondhe) [inline src/app/sb-admin/components/login/login.component.ts]: remove extra blank lines, keep only one @siddhishinde0723

File: src/config/url.config.json [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
+		"ORGANIZATION_UPDATE_URL": "api/org/v1/update",
+		"ORGNIZATION_SEARCH_URL": "api/org/v1/search",
+		"USER_CREATE_URL": "api/user/v3/create",
+		"USER_UPDATE_ROLE_URL": "api/user/v1/role/assign",
+		"USER_BLOCK_URL": "api/user/v1/block",
+		"USER_UNBLOCK_URL": "api/user/v1/unblock",
+		"USER_ROLES_GET_URL": "api/data/v1/form/read",
+		"USER_SEARCH_URL": "api/user/v1/search",
+		"COMPOSITE_SEARCH_URL": "action/composite/v3/search",
+		"GENRATE_TOKEN": "auth/realms/sunbird/protocol/openid-connect/token"
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/config/url.config.json]: fix this @siddhishinde0723

File: src/config/constant.config.ts [hunk 5]
```diff
@@ -16,4 +16,25 @@ export const SystemRoles = [
...
     { name: 'PUBLIC', role: 'Public', count: 0 }
+]
+
+export const SessionStorageKeys = {
+    ACCESS_TOKEN: 'access_token',
+    AUTH_TOKEN: 'auth_token',
+    TARGET_URL: 'target_url',
+  };
+
+  export const CategoryName = [
```
Comment (manojLondhe) [inline src/config/constant.config.ts]: fix indent @siddhishinde0723

File: src/locales/en/en.json [hunk 6]
```diff
@@ -142,6 +142,17 @@
   "COMMON_USERS":"User",
   "COMMON_YES":"Yes",
   "COMMON_NO":"No",
+  "LABEL_HOME":"Home",
+  
+	"ADD_AUTH": "Add Auth Token",
```
Comment (manojLondhe) [inline src/locales/en/en.json]: fix indent @siddhishinde0723

File: src/locales/en/en.json [hunk 7]
```diff
@@ -142,6 +142,17 @@
...
+	"ADD_USERNAME": "Enter Username",
```
Comment (manojLondhe) [inline src/locales/en/en.json]: add screenanme/entity-name as prefix to all lang strings
 @siddhishinde0723 

eg: ADD_USERNAME_INVALID => LOGIN_ADD_USERNAME_INVALID

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/components/login/login.component.html [hunk 8]
```diff
@@ -0,0 +1,101 @@
+<div class="login-container">
+	<div class="login-header">
+		<h2>Login</h2>
```
Comment (gouravmore) [inline src/app/sb-admin/components/login/login.component.html]: Use language constant for Login

File: src/app/sb-admin/components/login/login.component.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    const { authToken, targetURL } = updatedFormValues;
+    const sanitizedTargetUrl = targetURL.endsWith('/')
+      ? targetURL.slice(0, -1)
+      : targetURL;
+    this.sessionStorageService.setAuthToken(authToken);
+    this.sessionStorageService.setTargetUrl(sanitizedTargetUrl);
+
+    const body = this.createRequestBody(updatedFormValues, sanitizedTargetUrl);
+
+    this.http
```
Comment (gouravmore) [inline src/app/sb-admin/components/login/login.component.ts]: This should call from login service

File: src/locales/en/en.json [hunk 10]
```diff
@@ -142,6 +142,17 @@
   "COMMON_USERS":"User",
   "COMMON_YES":"Yes",
   "COMMON_NO":"No",
+  "LABEL_HOME":"Home",
+  
+	"LOGIN_ADD_AUTH": "Add Auth Token",
```
Comment (gouravmore) [inline src/locales/en/en.json]: indentation should be proper

File: src/app/layout/app.topbar.component.html [hunk 11]
```diff
@@ -10,9 +10,9 @@ <h4><b>{{'DASHBOARD_NAME'|i18next}}</b></h4>
         <i class="pi pi-ellipsis-v"></i>
     </button>
     <div #topbarmenu class="layout-topbar-menu" routerLink="/auth/login" [ngClass]="{'layout-topbar-menu-mobile-active': layoutService.state.profileSidebarVisible}">
-        <button class="p-link layout-topbar-button">
-            <i class="pi pi-user"></i>
-            <span>Profile</span>
+        <button (click)="logout()" class="p-link layout-topbar-button">
+            <i class="pi pi-power-off"></i>
+            <span>Logout</span>
```
Comment (gouravmore) [inline src/app/layout/app.topbar.component.html]: Use language constants for all the hard coded texts

**Review by gouravmore — APPROVED** [review]

**Pull request merged**